adapters/dynamoDB_adapters/src/main/java/org/atomhopper/dynamodb/adapter/DynamoDBFeedSource.java

```diff
@@ -866,13 +870,31 @@ private void addFeedCurrentLink(Feed hydratedFeed, String baseFeedUri) {
 
     private Entry hydrateEntry(PersistedEntry persistedEntry, Abdera abderaReference) {
 
-        final Document<Entry> hydratedEntryDocument = abderaReference.getParser().parse(
-                new StringReader(persistedEntry.getEntryBody()));
+//        final Document<Entry> hydratedEntryDocument = abderaReference.getParser().parse(
+//                new StringReader(persistedEntry.getEntryBody()));
+
+        String entryBody = persistedEntry.getEntryBody();
+        String contentType;
+        if(entryBody.trim().startsWith("{")){
+            contentType = "application/json";
+        }
+        else{
+            contentType = "application/atom+xml";
+        }
+
+        if (parser == null) {
+            LOG.error("Making sure parser is injected.");
+            throw new IllegalStateException("Parser was not injected into DynamoDB");
+        }
```

Comment:
The content type should be in a header and not be discovered from the body content. Do not parse the body to determine this, exclusively use the headers from the http request.